configs/azure_iot_sdksFunctions.cmake

```diff
@@ -1,6 +1,30 @@
 #Copyright (c) Microsoft. All rights reserved.
 #Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+function(getIoTSDKVersion)
+	# First find the applicable line in the file
+	file (STRINGS "${CMAKE_SOURCE_DIR}/iothub_client/inc/iothub_client_version.h" iotsdkverstr
+		REGEX "^[\t ]*#[\t ]*define[\t ]*IOTHUB_SDK_VERSION[\t ]*\"([0-9]+)[.]([0-9]+)[.]([0-9]+)\"")
+		
+	if (!MATCHES)
+		message(FATAL_ERROR "Unable to find version in ${CMAKE_SOURCE_DIR}/iothub_client/inc/iothub_client_version.h")
+	else(!MATCHES)
```

Comment:
This seem peculiar to me the if and the else both have !MATCHES.  I think you want the else just to haven and empty () after it.

Reply:
CMake allows one to put the condition statement on the else, and endif. In my opinion this is useful because it will match these up and if one, for example, misses an endif in a later code modification it will flag the mismatch of the conditional block's start and end making the omission easier to find. However, if you really hate it, I'll take it out. Your call.

---
In reply to: [140001036](https://github.com/Azure/azure-iot-sdk-c/pull/233#discussion_r140001036) [](ancestors = 140001036)